app.go

```diff
@@ -26,14 +26,27 @@ func app(cmd *cobra.Command, args []string) {
 	var err error
 
 	// Get version of Docker benchmark to run
-	if dockerVersion != "" {
-		version = dockerVersion
-	} else {
-		version, err = getDockerVersion()
-		if err != nil {
+	if benchmarkVersion != "" {
+		if dockerVersion != "" {
 			util.ExitWithError(
-				fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
-					err))
+				fmt.Errorf("Version check failed: Error can't specify both --version and --benchmark flags\nIf not specify neither docker version will be auto detect"))
+		}
+		version = benchmarkVersion
+	} else {
+		if dockerVersion != "" {
+			version, err = getDockerCisVersion(dockerVersion)
+			if err != nil {
+				util.ExitWithError(
+					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
+						err))
```

Comment:
This error message is misleading - in this case the user has supplied a version with `--version` but it couldn't be converted to a CIS benchmark version

```suggestion
				fmt.Errorf("Failed to get a valid CIS benchmark version for Docker version %s: %v",
					dockerVersion, err))
```